scripts/binary/binary-sources.js

```diff
@@ -0,0 +1,38 @@
+const fs = require('fs')
+const crypto = require('crypto')
+const path = require('path')
+
+const escapeString = (string) => string.replaceAll(`\``, `\\\``).replaceAll(`$`, `\\$`)
+
+function read ({ file }) {
+  const pathToFile = require.resolve(`./${file}`)
+
+  return fs.readFileSync(pathToFile, 'utf8')
+}
+
+const getBinaryEntryPointSource = () => {
+  return read({ file: 'binary-entry-point-source.js' })
+}
+
+const getIntegrityCheckSource = (baseDirectory) => {
+  const fileSource = read({ file: 'binary-integrity-check-source.js' })
+  const secret = require('crypto').randomBytes(48).toString('hex')
+
+  const mainIndexHash = crypto.createHmac('md5', secret).update(fs.readFileSync(path.join(baseDirectory, './index.js'), 'utf8')).digest('hex')
+  const bytenodeHash = crypto.createHmac('md5', secret).update(fs.readFileSync(path.join(baseDirectory, './node_modules/bytenode/lib/index.js'), 'utf8')).digest('hex')
+  const indexJscHash = crypto.createHmac('md5', secret).update(fs.readFileSync(path.join(baseDirectory, './packages/server/index.jsc'), 'utf8')).digest('hex')
+
+  return fileSource.split('\n').join(`\n  `)
+  .replaceAll('MAIN_INDEX_HASH', mainIndexHash)
+  .replaceAll('BYTENODE_HASH', bytenodeHash)
+  .replaceAll('INDEX_JSC_HASH', indexJscHash)
+  .replaceAll('HMAC_SECRET', secret)
```

Comment:
The step where we begin signing our sources and actively working against people trying to hack (in the original sense of "play around with") our open source project is a step too far, IMO.

I assume there was a product meeting and decisions around this and it's too late to change anyone's mind, but it does leave a sour taste in my mouth regardless.

Reply:
I see you have concerns here Blue. Let's continue this discussion offline with the product team.